signatures/pdf_sign_external_globalsign.go

```diff
@@ -0,0 +1,539 @@
+/*
+ * This example showcases how to digitally sign a PDF file using an external
+ * signing service which returns PKCS7 package. The external service is
+ * is simulated by signing the file with UniDoc.
+ *
+ * $ ./pdf_sign_external_globalsign <INPUT_PDF_PATH> <OUTPUT_PDF_PATH> <API_KEY> <API_SECRET> <CERT_FILE_PATH> <KEY_FILE_PATH>
+ */
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/unidoc/globalsign-dss"
+	"github.com/unidoc/pkcs7"
+	"github.com/unidoc/unipdf/v3/annotator"
+	"github.com/unidoc/unipdf/v3/common/license"
+	"github.com/unidoc/unipdf/v3/core"
+	"github.com/unidoc/unipdf/v3/model"
+
+	"golang.org/x/crypto/ocsp"
+)
+
+func init() {
+	// Make sure to load your metered License API key prior to using the library.
+	// If you need a key, you can sign up and create a free one at https://cloud.unidoc.io
+	err := license.SetMeteredKey(os.Getenv(`UNIDOC_LICENSE_API_KEY`))
+	if err != nil {
+		panic(err)
+	}
+}
+
+var (
+	sigLen = 8192
+
+	apiKey, apiSecret, certFilepath, keyFilepath string
+)
+
+const usagef = "Usage: %s INPUT_PDF_PATH OUTPUT_PDF_PATH API_KEY API_SECRET CERT_FILE_PATH KEY_FILE_PATH\n"
+
+func main() {
+	args := os.Args
+	if len(args) < 7 {
+		fmt.Printf(usagef, os.Args[0])
+		return
+	}
+	inputPath := args[1]
+	outputPath := args[2]
+
+	apiKey = args[3]
+	apiSecret = args[4]
+	certFilepath = args[5]
+	keyFilepath = args[6]
+
+	// This would be the time to send the PDF buffer to a signing device or
```

Comment:
This comment is not really relevant for this example. At least the part of the comment which states that the signature is simulated should be removed as the file is signed using the GlobalSign DSS client.

Reply:
Updated at 15bb7144b66b8891f5c68bd64803bb2d9ee32f54